API/hermes_abi/HermesABIHelpers.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#ifndef HERMES_ABI_HERMES_ABI_HELPERS_H
+#define HERMES_ABI_HERMES_ABI_HELPERS_H
+
+#include "hermes_abi/hermes_abi.h"
+
+namespace facebook::hermes::abi {
+
+#define DECLARE_HERMES_ABI_POINTER_HELPERS(name)                             \
+  static inline HermesABI##name create##name(HermesABIManagedPointer *ptr) { \
```

Comment:
It is better to keep the C-API header clean from any implementation code.
If such code is needed for C++ implementation, then it is better to put it in a separate C++ header which includes the C-API header.
We must be able to use the C API from any other language.